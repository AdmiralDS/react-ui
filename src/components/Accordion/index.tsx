import type { FC, HTMLAttributes, KeyboardEvent } from 'react';
import { useCallback, useRef } from 'react';

import { keyboardKey } from '../common/keyboardKey';
import { moveFocus, nextItem, previousItem } from './utils';
import { AccordionWrapper } from './style';

type Dimension = 'l' | 'm';

export * from './AccordionItem';

export interface AccordionProps extends HTMLAttributes<HTMLDivElement> {
  /** Размер компонента */
  dimension?: Dimension;
  /** Расположение иконки шеврона в заголовке. По умолчанию иконка выравнивается по правому краю. */
  iconPosition?: 'right' | 'left';
  /** Скрыть верхний разделитель аккордиона (верхнюю серую полоску).
   * Параметр hideTopDivider скрывает верхнюю серую полоску только у первого AccordionItem, заданного внутри Accordion
   */
  hideTopDivider?: boolean;
  /** Скрыть нижний разделитель аккордиона (нижнюю серую полоску)
   * Параметр hideBottomDivider скрывает нижнюю серую полоску только у последнего AccordionItem, заданного внутри Accordion
   */
  hideBottomDivider?: boolean;
}

export const Accordion: FC<AccordionProps> = ({
  children,
  dimension = 'l',
  iconPosition = 'right',
  hideTopDivider = false,
  hideBottomDivider = false,
  onKeyDown,
  ...props
}) => {
  const accordionRef = useRef<HTMLDivElement | null>(null);
  const handleKeyDown = useCallback((e: KeyboardEvent<HTMLDivElement>) => {
    let focusedOption: any = ((accordionRef.current && accordionRef.current.ownerDocument) || document).activeElement;
    // если фокус находится на AccordionItem Title, то можно переключаться между accordion items с помощью ArrowDown, ArrowUp, Home, End
    if (focusedOption.id.indexOf('accordion_title') > -1 && focusedOption.tagName.toLowerCase() === 'div') {
      do {
        focusedOption = focusedOption?.parentNode;
      } while (focusedOption.parentNode !== accordionRef.current);

      const code = keyboardKey.getCode(e);

      if (code === keyboardKey.ArrowDown) {
        moveFocus(accordionRef.current, focusedOption, nextItem);
        e.preventDefault();
      } else if (code === keyboardKey.ArrowUp) {
        moveFocus(accordionRef.current, focusedOption, previousItem);
        e.preventDefault();
      } else if (code === keyboardKey.Home) {
        e.preventDefault();
        moveFocus(accordionRef.current, null, nextItem);
      } else if (code === keyboardKey.End) {
        e.preventDefault();
        moveFocus(accordionRef.current, null, previousItem);
      }
    }
    onKeyDown?.(e);
  }, []);
  return (
    <AccordionWrapper
      ref={accordionRef}
      data-dimension={dimension}
      data-icon={iconPosition}
      $hideTopDivider={hideTopDivider}
      $hideBottomDivider={hideBottomDivider}
      $dimension={dimension}
      onKeyDown={handleKeyDown}
      {...props}
    >
      {children}
    </AccordionWrapper>
  );
};

Accordion.displayName = 'Accordion';
