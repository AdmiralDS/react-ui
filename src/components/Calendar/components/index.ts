export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
