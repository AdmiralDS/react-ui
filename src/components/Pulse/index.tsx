import styled, { keyframes, css } from 'styled-components';

export type PulseDimension = 'l' | 'm' | 's';
export type PulseStatus = 'info' | 'danger' | 'success' | 'warning';

export interface PulseProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Размер компонента */
  dimension?: PulseDimension;
  /** Статус компонента (цветовое окрашивание).
   * Можно задать кастомный цвет через объект со свойством background */
  status?: PulseStatus | { background?: string };
  /** Позволяет добавлять  миксин созданный с помощью styled css  */
  cssMixin?: ReturnType<typeof css>;
}

const AnimationS = keyframes`
 0% {
    width: 6px;
    height: 6px;
    opacity: 100%;
    border-width: 1px;
    filter: blur(0.33px);
  }

  80% {
    width: 20px;
    height: 20px;
    opacity: 0%;
    border-width: 1.5px;
    filter: blur(0.66px);
  }

  100% {
    opacity: 0%;
  }
`;

const AnimationM = keyframes`
 0% {
    width: 10px;
    height: 10px;
    opacity: 100%;
    border-width: 1px;
    filter: blur(0.33px);
  }
  80% {
    width: 28px;
    height: 28px;
    opacity: 0%;
    border-width: 2px;
    filter: blur(1px);
  }
  100% {
    opacity: 0%;
  }
`;

const AnimationL = keyframes`
 0% {
    width: 14px;
    height: 14px;
    opacity: 100%;
    border-width: 1px;
    filter: blur(0.33px);
  }
  80% {
    width: 36px;
    height: 36px;
    opacity: 0%;
    border-width: 3px;
    filter: blur(1.33px);
  }
  100% {
    opacity: 0%;
  }
`;

const getPulseColor = css<{ $status: PulseStatus | { background?: string } }>`
  ${({ theme, $status }) => {
    switch ($status) {
      case 'danger':
        return `var(--admiral-color-Error_Error60Main, ${theme.color['Error/Error 60 Main']})`;
      case 'success':
        return `var(--admiral-color-Success_Success50Main, ${theme.color['Success/Success 50 Main']})`;
      case 'warning':
        return `var(--admiral-color-Warning_Warning50Main, ${theme.color['Warning/Warning 50 Main']})`;
      case 'info':
        return `var(--admiral-color-Primary_Primary60Main, ${theme.color['Primary/Primary 60 Main']})`;
      default:
        return (
          $status?.background || `var(--admiral-color-Primary_Primary60Main, ${theme.color['Primary/Primary 60 Main']})`
        );
    }
  }}
`;

const PulseElement = styled.div<{
  $dimension: PulseDimension;
  $status: PulseStatus | { background?: string };
  $cssMixin?: ReturnType<typeof css>;
}>`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: ${(p) => (p.$dimension == 'l' ? 16 : p.$dimension == 'm' ? 12 : 8)}px;
  inline-size: ${(p) => (p.$dimension == 'l' ? 16 : p.$dimension == 'm' ? 12 : 8)}px;
  border-radius: 50%;
  background-color: ${getPulseColor};

  &:before {
    position: absolute;
    content: '';
    background-color: transparent;
    border-radius: 50%;
    border-style: solid;
    border-color: ${getPulseColor};
    box-sizing: border-box;
    animation-name: ${(p) => (p.$dimension == 'l' ? AnimationL : p.$dimension == 'm' ? AnimationM : AnimationS)};
    animation-duration: 2500ms;
    animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
    animation-iteration-count: infinite;
  }

  ${(p) => p.$cssMixin}
`;

export const Pulse: React.FC<PulseProps> = ({ dimension = 'm', status = 'info', cssMixin, ...props }) => {
  return <PulseElement $dimension={dimension} $status={status} $cssMixin={cssMixin} {...props} />;
};
