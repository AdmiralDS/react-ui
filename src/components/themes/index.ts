export * from './light';
export * from './dark';
export type { Theme } from './common';
