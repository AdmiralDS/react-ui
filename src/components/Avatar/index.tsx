import { forwardRef } from 'react';

import type { AvatarBaseInternalProps, AvatarBaseProps } from '#src/components/AvatarBase';
import { AvatarBase } from '#src/components/AvatarBase';

export interface AvatarProps extends Omit<AvatarBaseProps, 'withActivityRing' | 'showActivityRing'> {}

export const Avatar = forwardRef<HTMLButtonElement, AvatarProps & AvatarBaseInternalProps>(
  ({ dimension, ...props }: AvatarProps & AvatarBaseInternalProps, ref) => {
    return (
      <AvatarBase
        ref={ref}
        {...props}
        dimension={props.status && dimension === 'xxs' ? 'xs' : dimension}
        withActivityRing={false}
        showActivityRing={false}
      />
    );
  },
);

Avatar.displayName = 'Avatar';
