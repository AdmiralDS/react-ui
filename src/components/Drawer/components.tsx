import type { FC, HTMLAttributes } from 'react';
import { createContext, useContext, useLayoutEffect, useRef } from 'react';
import styled from 'styled-components';

import { typography } from '#src/components/Typography';
import { debounce } from '#src/components/common/utils/debounce';
import { resizePaddings } from '../common/dom/resizePaddings';

const Title = styled.h5<{ $mobile: boolean; $displayCloseIcon: boolean }>`
  ${({ $mobile }) => ($mobile ? typography['Header/H6'] : typography['Header/H5'])};
  color: var(--admiral-color-Neutral_Neutral90, ${(p) => p.theme.color['Neutral/Neutral 90']});
  margin: 0;
  padding: ${({ $mobile, $displayCloseIcon }) => {
    if ($mobile) {
      return $displayCloseIcon ? '0 48px 8px 16px' : '0 16px 8px';
    }
    return $displayCloseIcon ? '0 56px 8px 24px' : '0 24px 8px';
  }};
`;

const Content = styled.div<{ $mobile: boolean }>`
  overflow-y: auto;
  outline: none;
  scrollbar-gutter: stable;
  padding-block: 8px;
  padding-inline: ${({ $mobile }) => `${$mobile ? 16 : 24}px`};
  height: 100%;
`;

const ButtonPanel = styled.div<{ $mobile: boolean }>`
  display: flex;
  flex-direction: ${({ $mobile }) => ($mobile ? 'column-reverse' : 'row-reverse')};
  padding: ${({ $mobile }) => ($mobile ? '16px 16px 0' : '16px 24px 0')};

  & > button {
    margin: ${({ $mobile }) => ($mobile ? '0 0 16px 0' : '0 16px 0 0')};
    ${({ $mobile }) => $mobile && 'width: 100%;'}
  }

  & > button:first-child {
    margin: 0;
  }
`;

export const DrawerContext = createContext({ mobile: false, displayCloseIcon: true } as {
  mobile: boolean;
  displayCloseIcon: boolean;
});

export const DrawerTitle: FC<HTMLAttributes<HTMLHeadingElement>> = ({ children, ...props }) => {
  const { mobile, displayCloseIcon } = useContext(DrawerContext);
  const asProp = mobile ? 'h6' : 'h5';
  return (
    <Title $mobile={mobile} $displayCloseIcon={displayCloseIcon} as={asProp} {...props}>
      {children}
    </Title>
  );
};

export const DrawerContent: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const contentRef = useRef<HTMLDivElement | null>(null);
  const mobile = useContext(DrawerContext).mobile;

  useLayoutEffect(() => {
    const node = contentRef.current;
    if (node) {
      resizePaddings(node);
      const resizeObserver = new ResizeObserver(debounce(() => resizePaddings(node), 250));
      resizeObserver.observe(node);
      return () => {
        resizeObserver.unobserve(node);
      };
    }
  }, []);

  return (
    <Content tabIndex={-1} ref={contentRef} $mobile={mobile} {...props}>
      {children}
    </Content>
  );
};

export const DrawerButtonPanel: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const mobile = useContext(DrawerContext).mobile;
  return (
    <ButtonPanel $mobile={mobile} {...props}>
      {children}
    </ButtonPanel>
  );
};
