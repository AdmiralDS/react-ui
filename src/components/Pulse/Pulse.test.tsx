import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { LIGHT_THEME } from '#src/components/themes';

import { Pulse } from '.';

describe('Pulse', () => {
  it('should render default component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Pulse />
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with different dimensions', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Pulse dimension="l" />
        <Pulse dimension="m" />
        <Pulse dimension="s" />
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with different statuses', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Pulse status="info" />
        <Pulse status="danger" />
        <Pulse status="success" />
        <Pulse status="warning" />
        <Pulse status={{ background: '#8A3FFC' }} />
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
