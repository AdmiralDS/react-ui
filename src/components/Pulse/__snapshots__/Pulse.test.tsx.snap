// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulse should render component with different dimensions 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 16px;
  inline-size: 16px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c0:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: IezNe;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c1 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c1:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c2 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 8px;
  inline-size: 8px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c2:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: fXgQcW;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

<body>
    <div>
      <div
        class="c0"
      />
      <div
        class="c1"
      />
      <div
        class="c2"
      />
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 16px;
  inline-size: 16px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c0:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: IezNe;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c1 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c1:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c2 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 8px;
  inline-size: 8px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c2:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: fXgQcW;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

<div>
    <div
      class="c0"
    />
    <div
      class="c1"
    />
    <div
      class="c2"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pulse should render component with different statuses 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c0:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c1 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Error_Error60Main, #D92020);
}

.c1:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Error_Error60Main, #D92020);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c2 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Success_Success50Main, #1BA049);
}

.c2:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Success_Success50Main, #1BA049);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c3 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Warning_Warning50Main, #FF5C22);
}

.c3:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Warning_Warning50Main, #FF5C22);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c4 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: #8A3FFC;
}

.c4:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: #8A3FFC;
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

<body>
    <div>
      <div
        class="c0"
      />
      <div
        class="c1"
      />
      <div
        class="c2"
      />
      <div
        class="c3"
      />
      <div
        class="c4"
      />
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c0:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c1 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Error_Error60Main, #D92020);
}

.c1:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Error_Error60Main, #D92020);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c2 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Success_Success50Main, #1BA049);
}

.c2:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Success_Success50Main, #1BA049);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c3 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Warning_Warning50Main, #FF5C22);
}

.c3:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Warning_Warning50Main, #FF5C22);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

.c4 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: #8A3FFC;
}

.c4:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: #8A3FFC;
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

<div>
    <div
      class="c0"
    />
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
    <div
      class="c4"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Pulse should render default component 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c0:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  block-size: 12px;
  inline-size: 12px;
  border-radius: 50%;
  background-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
}

.c0:before {
  position: absolute;
  content: '';
  background-color: transparent;
  border-radius: 50%;
  border-style: solid;
  border-color: var(--admiral-color-Primary_Primary60Main, #0062FF);
  box-sizing: border-box;
  animation-name: cIwzhZ;
  animation-duration: 2500ms;
  animation-timing-function: cubic-bezier(0, 0, 0.58, 1);
  animation-iteration-count: infinite;
}

<div>
    <div
      class="c0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
